storage/postgres/query_builder.go

```diff
@@ -44,88 +43,95 @@ func NewQueryBuilder(db pgDB) *QueryBuilder {
 }
 
 // NewQuery constructs new queries for the current query builder db
-func (qb *QueryBuilder) NewQuery() *pgQuery {
+func (qb *QueryBuilder) NewQuery(entity PostgresEntity) *pgQuery {
```

Comment:
General Comment: This whole query building looks like it's getting quite complicated for us to support. How about we discuss if we could benefit from researching some library for queries or drop the abstraction layer altogether and directly use a sql-building library.

Reply:
Something like this?
https://github.com/Masterminds/squirrel